web/styles/app_components.css

```diff
@@ -575,10 +575,8 @@ div.overlay {
     float: right;
     padding: 0 4px;
     height: 16px;
-    line-height: 16px;
```

Comment:
@karlstolley are there interactions we should worry about with leaving the `height` as-is with changing the `line-height`? I just note that it looks intentional that those were the same.

Reply:
I agree that it looks intentional, but I also think it was incorrect. I'll pull down this PR and review what the computed value is, etc. and get back with a definitive answer tomorrow.

Reply:
Okay. I think the `line-height: 16px` value needs to stay for now; changing it here at least has unintended side-effects with the unread counts on streams and topics. I do think this is another case where we're trying to vertically center something in a tiny space, and there's an even/odd mismatch so that almost any adjustment--including this one--is going to make something too high or too low relative to center. 

Put another way, if you have, say, a 5px block that you're trying to center in an 8px space, there's either going to be 2px above the block (too low) or 2px below (too high).

That said, @shameondev, you ought to add a couple of adjustments here to let flexbox better position the unreads in the global filters area. Here's a screenshot of my adjustments from the dev tools. Basically we want to let flexbox handle the vertical centering of the unread count container (`align-self: center`), and to do that, we ought to zero out the `margin-top: 0`. I think we can also ditch the `!important` keyword there--have a look yourself, but going through the list of selectors there, I didn't see any place where `!important` was actually needed. But be sure to double-check that yourself:

<img width="382" alt="gf-adjustments" src="https://github.com/zulip/zulip/assets/170719/6388f62d-62fc-4665-9fa2-581e5f5cbf84">

Reply:
(It might be that `!important` is there to guard against selector order impacting whether `margin: 0` is applied, but I think `!important` introduces too many other problems to be worth saving.)

Reply:
Thanks for the explanation, @karlstolley. Everything seems reasonable. I suggest we fix only the horizontal alignment in this PR and leave the vertical aligning as a follow-up to #26468, where we introduced flex, then.

Reply:
I'll remove the second commit that removes the `line-height` property.

Reply:
In the follow-up I mentioned [here](https://chat.zulip.org/#narrow/stream/9-issues/topic/long.20names.20of.20.60global_filters.60.20break.20the.20layout/near/1624897), I already removed the `margin-top` property and will add the `align-self`.

Reply:
I have a fix that will be part of #24994 that allow us to remove the line-height here, too, for perfect centering on all the unread counts, up and down the left side bar (in short: it was a matter of fixing a goofy 17.5px line-height on the Direct Messages items). Everything else aligns much better without that value. We just needed to fix up this edge case.